DataPlotly/gui/layout_item_gui.py

```diff
@@ -36,50 +41,138 @@ def __init__(self, parent, layout_object):
         vl = QVBoxLayout()
         vl.setContentsMargins(0, 0, 0, 0)
 
-        self.plot_properties_button = QPushButton(self.tr('Setup Plot'))
-        vl.addWidget(self.plot_properties_button)
-        self.plot_properties_button.clicked.connect(self.show_properties)
+        plot_tools_layout = QHBoxLayout()
+
+        plot_add_button = QPushButton(self.tr('Add Plot'))
+        plot_add_button.setToolTip('Add a new plot')
+        plot_tools_layout.addWidget(plot_add_button)
+        plot_add_button.clicked.connect(self.add_plot)
+
+        plot_remove_button = QPushButton(self.tr('Remove Plot'))
+        plot_remove_button.setToolTip('Remove selected plot')
+        plot_tools_layout.addWidget(plot_remove_button)
+        plot_remove_button.clicked.connect(self.remove_plot)
+
+        plot_move_up_button = QPushButton(self.tr('Move Up'))
+        plot_move_up_button.setToolTip('Move selected plot up')
+        plot_tools_layout.addWidget(plot_move_up_button)
+        plot_move_up_button.clicked.connect(self.move_up_plot)
+
+        plot_move_down_button = QPushButton(self.tr('Move Down'))
+        plot_move_down_button.setToolTip('Move selected plot down')
+        plot_tools_layout.addWidget(plot_move_down_button)
+        plot_move_down_button.clicked.connect(self.move_down_plot)
+
+        vl.addLayout(plot_tools_layout)
+
+        self.plot_list = QListWidget()
+        self.plot_list.setSelectionMode(QListWidget.SingleSelection)
+        vl.addWidget(self.plot_list)
+        self.populate_plot_list()
+
+        plot_properties_button = QPushButton(self.tr('Setup Selected Plot'))
+        vl.addWidget(plot_properties_button)
+        plot_properties_button.clicked.connect(self.show_properties)
 
         self.panel = None
         self.setPanelTitle(self.tr('Plot Properties'))
         self.item_properties_widget = QgsLayoutItemPropertiesWidget(self, layout_object)
         vl.addWidget(self.item_properties_widget)
         self.setLayout(vl)
 
+    def populate_plot_list(self):
+        """
+        Clears and re-populates the plot list widget. The currently selection is retained
+        """
+        selected_index = self.plot_list.currentRow()
+        self.plot_list.clear()
+        for setting in self.plot_item.plot_settings:
+            plot_type = setting.plot_type + ' ' if setting.plot_type is not None else '(not set)'
+            legend_title = setting.properties.get('name', '')
+            self.plot_list.addItem(plot_type + ' [' + legend_title + ']')
```

Comment:
Should this also omit the [ ] if legend_title is empty?

Reply:
I have removed the [ ] if legend_title is empty